### DISCUSSION TITLE
Nullable pointers

### DISCUSSION THREAD

**Discussion by Happypig375** [dotnet/csharplang#3307] [Language Ideas]
Now that in C# 8 we have nullable reference types, I started annotating an existing library. However, I discovered that pointers cannot be annotated:
```cs
public byte*? dist_context_map = null; // Error CS1519: Invalid token '?' in class, struct, or interface member declaration
```
Why is this the case? Can't we enable nullables all the way?

**Comment by Unknown6656** [comment]
A pointer is effectively stored as a `int32` or `int64` on the stack, meaning that it is a struct. The nullable refrence type feature of C#8 would hence be of not much use. However, you can do the following:
```csharp
byte* ptr = null;
// is the same as:
byte* ptr = (byte*)0;

if (ptr == null) // "ptr is null" does also work here
{
    // do something
}
```

If using `null` on `T*` is not a valid solution for you, you might resort to `IntPtr?` and pointer casts

**Comment by NN---** [comment]
Did you mean you want non nullable pointers ?
